course/management/commands/delete_instance.py

```diff
@@ -0,0 +1,53 @@
+from django.core.management.base import BaseCommand
+from course.models import CourseInstance
+import datetime
+
+class Command(BaseCommand):
+    help = "Delete course instance along with related data (exercises, submissions, etc)."
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '-c',
+            '--course-instance',
+            type=int,
+            help='Course instance ID to be deleted.',
+        )
+        parser.add_argument(
+            '-e',
+            '--ending-since',
+            type=int,
+            help='Delete all course instances that ended since N days ago.',
+        )
+        parser.add_argument(
+            '--dry-run',
+            action='store_true',
+            help='Do not delete anything, but show which instances would be deleted.',
+        )
+
+    def handle(self, *args, **options):
+        course_instances: list[CourseInstance] = []
+        
+        if options['ending_since']:
+            deletetime = datetime.datetime.now(datetime.timezone.utc) - datetime.timedelta(
+                days=options['ending_since']
+            )
+            course_instances = list(CourseInstance.objects.filter(
+                ending_time__lt=deletetime
+            ))
+        
+        if options['course_instance']:
+            try:
+                course_instances.append(CourseInstance.objects.get(id=options['course_instance']))
+            except CourseInstance.DoesNotExist:
+                self.stderr.write('Course instance {} does not exist'.format(options['course_instance']))
+
+        if options['dry_run']:
+            self.stdout.write('This is just a dry run. Nothing will be deleted.')
+
+        for c in course_instances:
+            self.stdout.write("deleting instance {} from course {}".format(
```

Comment:
The model classes define `__str__` so that `str(c)` gives a nice string description.

Reply:
fixed.